Help & Support: Add the app log feature

mindgraffiti opened the pull request:
Closes #618.

This PR adds the ability to view, share, and send application logs. 

## To Test
1. Log in and navigate to My store > Settings > Help & Support > View Application Log.
2. Select any log available and view it.
3. Select the share button and send it, save it, or share it. Verify it arrived at the expected place.
4. When ready, select the "Clear old activity logs" button. Verify only the "Current" row and application log remains.

<img src="https://user-images.githubusercontent.com/1062444/51944148-e3065400-23e0-11e9-9112-ab0ffcc539b8.gif" width="300" />

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

cc: @bummytime or @astralbodies first review wins 😀

mindgraffiti commented on WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogDetailViewController.swift:
```diff
@@ -0,0 +1,86 @@
+        self.logDate = ""
+        super.init(coder: aDecoder)
+    }
+
```
Initializers on a view controller isn't something I'm used to creating. Are these correct?

mindgraffiti commented on WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogViewController.swift:
```diff
@@ -0,0 +1,264 @@
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let vc = segue.destination as? ApplicationLogDetailViewController, segue.identifier == Segues.detailSegue {
+            let logFileInfo = sender as! DDLogFileInfo
```
Getting the file contents here also seems weird. Is there a better place for this?

astralbodies commented on WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogDetailViewController.swift:
```diff
@@ -0,0 +1,86 @@
+    /// Specifies the activity types that should be excluded.
+    /// - returns: all supported types except `.copyToPasteboard` and `.mail`.
+    ///
+    func assembleExcludedSupportTypes() -> [UIActivity.ActivityType] {
```
I honestly don't remember why I added this - probably a good ol' copy and paste from StackOverflow.

astralbodies replied:
Could be a nice simple PR back to WPiOS to remove it from there too!

mindgraffiti replied:
I think we should restore the ability to email it, too, since the No. 1 reason someone is probably looking at their application logs is because an HE asked for it to be sent.

astralbodies commented on WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogViewController.swift:
```diff
@@ -0,0 +1,273 @@
+    /// Clear old logs action
+    ///
+    func clearLogsWasPressed() {
+        for logFileInfo in logFiles {
```
Ooh I love this @bummytime - I *always* forget about `where`!

mindgraffiti commented:
@bummytime thank you for the fancy `where` clause hint. I fixed the problem with the activity sheet not displaying the ability to mail or copy logs. Also fixed the crash when deleting old logs. Ready for another review!

mindgraffiti commented:
Thanks Matt!
